mypy issue with latest version (0.800):

### Information

- **Qiskit IBMQ Provider version**: master
- **Python version**: 3.7 / 3.8
- **Operating system**: linux ubuntu focal

### What is the current behavior?

CI is failing with this:

>qiskit/providers/ibmq/experiment/experimentservice.py:347: error: Incompatible types in assignment (expression has type "Tuple[str, int]", variable has type "Union[str, ResultQuality]")  [assignment]

Looks like maybe that was picked up with the new 0.800 release of mypy?

https://mypy-lang.blogspot.com/2021/01/mypy-0800-released.html

### Steps to reproduce the problem

Seen here (and locally): https://github.com/Qiskit/qiskit-ibmq-provider/pull/866/checks?check_run_id=1763742656

### What is the expected behavior?

mypy should only fail on newly introduced issues, not latent ones

### Suggested solutions

Fix or add ignore for the issue, or cap mypy if 0.800 is problematic.

--- a/qiskit/providers/ibmq/experiment/experimentservice.py
+++ b/qiskit/providers/ibmq/experiment/experimentservice.py
@@ -335,7 +335,7 @@ class ExperimentService:
         if quality:
             for op, qual in quality:
                 if isinstance(qual, ResultQuality):
-                    qual = qual.value
+                    qual = qual.value  # type: ignore[assignment]
                 qual_str = qual if op == 'eq' else "{}:{}".format(op, qual)
                 qualit_list.append(qual_str)
         results = []

